packages/sanes-chrome-extension/src/routes/tx-request/components/ShowRequest/ReqRegisterUsernameTx.tsx

```diff
@@ -1,16 +1,74 @@
 import { RegisterUsernameTx } from '@iov/bns';
-import { List } from 'medulas-react-components/lib/components/List';
+import { List, ListItem, ListItemText } from 'medulas-react-components/lib/components/List';
 import * as React from 'react';
+import { makeStyles } from '@material-ui/core';
+import { amountToString } from '../../../../utils/balances';
+import Typography from 'medulas-react-components/lib/components/Typography';
+import Block from 'medulas-react-components/lib/components/Block';
+
+export const REQ_REGISTER_USERNAME = 'req-register-username-tx';
+
+const useStyles = makeStyles({
+  root: {
+    margin: 0,
+  },
+});
+
+const secondaryProps = {
+  noWrap: true,
+};
 
 interface Props {
   readonly creator: string;
   readonly tx: RegisterUsernameTx;
 }
 
-const ReqRegisterUsernameTx = ({ tx, creator }: Props): JSX.Element => {
-  // TODO: implement
-  // https://github.com/iov-one/ponferrada/issues/349
-  return <List />;
+const ReqRegisterUsernameTx = ({ tx }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  const listItemClasses = { root: classes.root };
+
+  const txFee =
+    tx.fee && tx.fee.tokens ? (
+      <React.Fragment>
+        <Block marginTop={1} />
+        <Typography variant="body1" inline>
+          Fee:{' '}
+        </Typography>
+        <Typography variant="body1" inline>
+          {amountToString(tx.fee.tokens)}
+        </Typography>
+      </React.Fragment>
+    ) : (
+      undefined
+    );
+
+  return (
+    <React.Fragment>
+      <Typography variant="body1" inline color="primary" id={REQ_REGISTER_USERNAME}>
+        {tx.username}*iov
+      </Typography>
+      <Typography variant="body1" inline>
+        {' '}
+        username registration request.
+      </Typography>
+      <Block marginTop={1} />
+      <Typography variant="body1">Addresses:</Typography>
+      <List>
+        {tx.addresses.map(address => (
+          <ListItem key={address.address}>
+            <ListItemText
+              classes={listItemClasses}
+              primary={address.chainId}
```

Comment:
@apanizo already created a chain ID -> chain name mapping (configured in config files). Can we easily use this here, showing the chain name? If that is too complicated, feel free to postpone for after the MvP

Reply:
@apanizo can you provide any code example how to use this in react component.